tds.go

```diff
@@ -50,12 +50,10 @@ func parseInstances(msg []byte) map[string]map[string]string {
 	return results
 }
 
-func getInstances(ctx context.Context, address string) (map[string]map[string]string, error) {
+func getInstances(ctx context.Context, p connectParams) (map[string]map[string]string, error) {
 	maxTime := 5 * time.Second
```

Comment:
With this PR, is maxTime still needed?

Reply:
Yes, on line 61 it uses it for setting the deadline on the connection

Reply:
Sorry, line 61 was in the "fold" for me and the scope and I didn't notice I was one `}` short  (meaning that I hadn't read to the end of `getInstances` yet).